[DESKTOP-1894] Add Jenkinsfile

StefanScherer opened the pull request:
This PR adds a Jenkinsfile to build, test, package and later also publish/push the package.

mikeparker commented:
I don't know much about the format but it looks fine to me. I suppose a few basic questions:
1. Whats the quiet period?
2. Do we need to keep logs for 30 days?
3. What does preserve stashes do
4. Is a 1 hour timeout a little long? (it should take like 5 minutes)
5. Do we know that the selected image supports building .net standard .net core and .net framework packages?

StefanScherer commented:
Thanks @mikeparker for your review and questions.
I've changed some settings in the Jenkinsfile.
1. removed the quiet period. This is for more heavier pipelines when a lot of commits arrive, wait for the last commit and then run a build after the quiet period.
2. keep only the last 10 logs
3. I've removed the preserved stashes, not needed in that pipeline.
4. well that's a timeout, I don't want to have it too short. Spinning up the windows-2019 instance and pulling the base image will take longer than 5 minutes when there is no instance running. Once it's up and running, another build will be much faster. So, keep it at 1 hour for now.
5. I don't know. I think we have to try it out to see if the whole pipeline works after moving to Jenkins.

StefanScherer commented:
The Jenkins pipeline is here: https://ci.docker.com/public/job/HttpOverStream/job/master/
